echopype/clean/utils.py

```diff
@@ -83,6 +84,95 @@ def calc_transient_noise_pooled_Sv(
     return pooled_Sv
 
 
+def index_binning_pool_Sv(
+    ds_Sv: xr.Dataset,
+    func: str,
+    depth_bin: int,
+    num_side_pings: int,
+    exclude_above: float,
+    chunk_dict: dict,
+) -> xr.DataArray:
+    """
+    Compute pooled Sv array for transient noise masking using index binning.
+
+    This function makes the assumption that within each channel, the difference
+    between depth values is uniform across all pings. Thus, computing the number of
+    range sample indices needed to cover the depth bin is a channel-specific task.
+    """
+    # Create `ds_Sv` copy
+    ds_Sv_copy = (
+        ds_Sv.copy()
```

Comment:
this is copy of the copy from the calling function (`mask_transient_noise`) -- is this needed?

Reply:
How about:
- remove the copy in the calling function, so only doing the copy within this function
- doing `range_sample=slice(min_range_sample, None)` along with the copy operation here

Reply:
Nah, I'll remove it